eo/src/eoRanking.h

```diff
@@ -40,73 +40,85 @@ template <class EOT>
 class eoRanking : public eoPerf2Worth<EOT> // false: do not cache fitness
 {
 public:
-
     using eoPerf2Worth<EOT>::value;
 
-  /* Ctor:
-   @param _p selective pressure (in (1,2]
-   @param _e exponent (1 == linear)
-  */
-  eoRanking(double _p=2.0, double _e=1.0):
-    pressure(_p), exponent(_e) {}
+    /* Ctor:
+     @param _p selective pressure (in (1,2]
+     @param _e exponent (1 == linear)
+    */
+    eoRanking(double _p = 2.0, double _e = 1.0) : pressure(_p), exponent(_e)
+    {
+        if (pressure <= 1.0)
+        {
+            std::string msg = "eoRanking: pressure must be > 1.0";
+            eo::log << eo::errors << "ERROR: " << msg << std::endl;
+            throw eoException(msg);
+        }
 
-  /* helper function: finds index in _pop of _eo, an EOT * */
-  int lookfor(const EOT *_eo, const eoPop<EOT>& _pop)
+        if (exponent > 2.0)
+        {
+            std::string msg = "eoRanking: exponent must be <= 2.0";
+            eo::log << eo::errors << "ERROR: " << msg << std::endl;
+            throw eoException(msg);
+        }
+    }
+
+    /* helper function: finds index in _pop of _eo, an EOT * */
+    int lookfor(const EOT *_eo, const eoPop<EOT> &_pop)
     {
-      typename eoPop<EOT>::const_iterator it;
-      for (it=_pop.begin(); it<_pop.end(); it++)
+        typename eoPop<EOT>::const_iterator it;
+        for (it = _pop.begin(); it < _pop.end(); it++)
         {
-          if (_eo == &(*it))
-            return it-_pop.begin();
+            if (_eo == &(*it))
+                return it - _pop.begin();
         }
-      throw eoException("Not found in eoLinearRanking");
+        throw eoException("Not found in eoLinearRanking");
     }
 
-  /* COmputes the ranked fitness: fitnesses range in [m,M]
-     with m=2-pressure/popSize and M=pressure/popSize.
-     in between, the progression depstd::ends on exponent (linear if 1).
-   */
-  virtual void operator()(const eoPop<EOT>& _pop)
+    /* COmputes the ranked fitness: fitnesses range in [m,M]
+       with m=2-pressure/popSize and M=pressure/popSize.
+       in between, the progression depstd::ends on exponent (linear if 1).
+     */
+    virtual void operator()(const eoPop<EOT> &_pop)
     {
-      std::vector<const EOT *> rank;
-      _pop.sort(rank);
-      unsigned pSize =_pop.size();
-      unsigned int pSizeMinusOne = pSize-1;
+        std::vector<const EOT *> rank;
+        _pop.sort(rank);
+        unsigned pSize = _pop.size();
+        unsigned int pSizeMinusOne = pSize - 1;
 
-      if (pSize <= 1)
-        throw eoPopSizeException(pSize,"cannot do ranking with population of size <= 1");
+        if (pSize <= 1)
+            throw eoPopSizeException(pSize, "cannot do ranking with population of size <= 1");
```

Comment:
This one is in the grey zone, for instance, given that some algorithms have the population size changing…